CRM/Core/Resources.php

```diff
@@ -881,4 +882,27 @@ private function resolveFileName(&$fileName, $extName) {
     }
   }
 
+  /**
+   * @param string $url
+   * @return string
+   */
+  public function addCacheCode($url) {
+    parse_str(parse_url($url, PHP_URL_QUERY), $queryParts);
+    $existing = isset($queryParts['r']) ? $queryParts['r'] : NULL;
+    $latest = $this->cacheCode;
+
+    if ($existing) {
+      if ($existing === $latest) {
+        return $url; // no need to update
+      }
+      else {
+        return str_replace('r=' . $existing, 'r=' . $latest, $url);
```

Comment:
OK, so this leads to my main confusion in the PR: I don't understand the scenario where (a) the URL already has `r=XYZ` and (b) we need to change it. Do you have an example?